### PR TITLE
test(ISD-281): Fix test-and-publish by setting setup-devstack-swift t…

### PR DESCRIPTION
**Pull request by amandahla**
Set setup-devstack-swift to True and add ```--openstack-rc=${GITHUB_WORKSPACE}/openrc``` to tox.

### CODE REVIEW

**Review by gtrkiller — APPROVED** [review]
LGTM

**Review by merkata — DISMISSED** [review]

File: .github/workflows/test_and_publish_charm.yaml [hunk 1]
```diff
@@ -18,3 +18,6 @@ jobs:
   publish-to-edge:
     uses: canonical/operator-workflows/.github/workflows/test_and_publish_charm.yaml@main
     secrets: inherit
+    with:
+      setup-devstack-swift: true
```
Comment (jdkandersson) [inline .github/workflows/test_and_publish_charm.yaml:22]: Is this always required for every charm?

**Review by jdkandersson — APPROVED** [review]

**Review by arturo-seijas — DISMISSED** [review]
LGTM. I know is out of scope but, can you pin the base docker image to `jammy`?

**Comment by amandahla** [comment]
> LGTM. I know is out of scope but, can you pin the base docker image to `jammy`?

Sure. Done.

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Pull request merged**